### PR TITLE
uibutton: add global nav button support

### PR DESCRIPTION
**Pull request by milas**
Can specify global nav buttons by passing `nav=True`.

### CODE REVIEW

**Review by ellenkorbes — APPROVED** [review]
works on my machine!

**Comment by milas** [comment]
Improved the `README` somewhat in 71ed8dc - I re-ordered it to keep the example near the top and added a section on icons with some tips and an example of using `read_file` to get an SVG from disk.

File: uibutton/README.md [hunk 1]
```diff
@@ -14,30 +14,85 @@ Creates a button for a resource that runs the given command when clicked.
...
-| `argv` | `List[str]` | Local command to run when button is clicked |
-| `text` | `str` | Text to display on button (optional: defaults to `name`) |
-
+| `name`      | `str`            | Unique ID for button |
+| `resource`  | `str`            | Resource to associate button with (required if `nav=False`) |
+| `argv`      | `List[str]`      | Local command to run when button is clicked |
+| `text`      | `str`            | Text to display on button (optional: defaults to `name`) |
+| `icon_name` | `str`            | Name of [Material Icons font ligature][material-icons-font] to use as icon |
+| `icon_svg`  |  `str` or `Blob` | `<svg>` to use as icon; should have 24x24px viewport |
+| `nav`       | `bool`           | If `True`, button will appear in global nav; otherwise, will be in resource view for specified resource |
```
[the remark this answers is not in the document]
Comment (milas) [reply uibutton/README.md]: Added an enum in a1cfa97 - I didn't make the entire exported API a struct for now - `load('ext://uibutton', 'cmd_button', 'location')` seems good enough/consistent with the other extensions. (Module seems nice for the future though)

File: uibutton/Tiltfile [hunk 2]
```diff
@@ -1,27 +1,56 @@
...
+
+  if not nav and not resource:
+    fail('Must provide a resource name or nav=True')
+
+  if not argv:
+    fail('argv cannot be empty')
+
+  btn_annotations = {}
+  if resource:
+    btn_annotations['tilt.dev/resource'] = resource
```
[the remark this answers is not in the document]
Comment (milas) [reply uibutton/Tiltfile]: Ehhh it wasn't _really_ but you could use it as a sneaky way to get a global nav button whose logs would show up in a specific resource, but that's probably more confusing than helpful, so I made this stricter with the `location` refactoring

**Review by landism — APPROVED** [review]

**Pull request merged**